WordPress/Sniffs/WP/TemplatedInlineScriptsSniff.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Sniffs\WP;
+
+use PHP_CodeSniffer\Exceptions\RuntimeException;
+use PHP_CodeSniffer\Util\Tokens;
+use PHPCSUtils\Tokens\Collections;
+use PHPCSUtils\Utils\TextStrings;
+use WordPressCS\WordPress\Sniff;
+
+/**
+ * Makes sure inline scripts and scripts are generated using either {@see wp_get_script_tag()} or {@see wp_get_inline_script_tag()}.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @since   3.0.0
+ */
+class TemplatedInlineScriptsSniff extends Sniff {
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		$targets   = Collections::$textStingStartTokens;
+		$targets[] = \T_INLINE_HTML;
+
+		return $targets;
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param int $stackPtr The position of the current token in the stack.
+	 *
+	 * @return void
+	 */
+	public function process_token( $stackPtr ) {
+
+		$content = $this->tokens[ $stackPtr ]['content'];
+		if ( \T_INLINE_HTML !== $this->tokens[ $stackPtr ]['code'] ) {
+			try {
+				$content = TextStrings::getCompleteTextString( $this->phpcsFile, $stackPtr );
+			} catch ( RuntimeException $e ) {
+				// Not the first token in a multi-line text string. Any issues will already have been reported.
+				return;
+			}
+		}
+
+		if ( preg_match_all( '#<script\b(?![^>]*\btext\/html\b)(?![^>]*\btext\/template\b)(?![^>]*\bapplication\/json\b)(?![^>]*\bsrc=(\\\'|\")\b)[^>]*>#', $content, $matches, PREG_OFFSET_CAPTURE ) > 0 ) {
```

Comment:
Why is the `'` escaped? This is the HTML context, right?

Reply:
It's a single quoted sting in PHP, so the `'` needs escaping for that.